### PR TITLE
Add tests of demes event timings.

### PR DESCRIPTION
**Pull request by molpopgen**
This PR adds tests and demes graphs that can also serve as documentation/examples

### CODE REVIEW

File: doc/long_vignettes/demes_event_timings.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+During some of our simulations, we will record the current time point
+and the part of the metadata containing individual parents.
+
+We can use a simple data class to record this information:
+
+
+```{literalinclude} ../../tests/test_demes_event_timings.py
+:language: python
+:lines: 8-20
```
Comment (apragsdale) [inline doc/long_vignettes/demes_event_timings.md:46]: Not pointing out anything wrong.. but I suppose hard-coding line numbers can be dangerous if the source file changes as all?

File: doc/long_vignettes/demes_event_timings.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  (thinking forwards in time).
+  During this interval, all individuals in `deme0` have parents from that same dime.
+  Likewise for `deme1`.
+* All individuals born in the pulse generation (10 generations ago) have all of their
+  parents drawn from `deme0`.
+  The reason for this is that there are no events affecting the ancestry of `deme0` and
+  the proportion of `deme1` ancestry that comes from `deme0` is 1.0, which is all of it.
+* The time interval $[11, \infty)$ generations ago is the *pre* pulse interval 
+  (thinking forwards in time).
+  During this interval, all individuals in `deme0` have parents from that same dime.
```
Comment (apragsdale) [inline doc/long_vignettes/demes_event_timings.md:129]: dime -> deme

**Review by apragsdale — COMMENTED** [review]
Review prior to viewing html of output.

File: doc/long_vignettes/demes_event_timings.md (hunk 1 above)
Comment (molpopgen) [reply doc/long_vignettes/demes_event_timings.md:46]: Yes, that is a problem.  Sadly, we just have t o be careful.

**Pull request merged**